app/components/PdfUploader.tsx

```diff
@@ -35,7 +35,10 @@ const PdfUploader: React.FC<Props> = ({ setPdfText, setIsLoading }) => {
         loadingTask.promise.then(
           (pdfDoc) => {
             const numPages = pdfDoc.numPages;
-            for(let i = 1; i <= Math.min(numPages, 5); i++) {
+            if (numPages > 4) {
+              setError('You uploaded a long PDF mate. Unfortunately only the first 4 pages will be consider since this is a free project.');
```

Comment:
I would make this alert a bit more formal if possible haha, chatgpt suggests sth like "Please note that due to the limitations of our free service, only the first four pages will be considered for processing."

It is funny though, so both are fine for me; but if you prefer the current version, I would fix the typo in "will be consider" to "will be considered"